ray-operator/apis/ray/v1/raycluster_types.go

```diff
@@ -158,6 +158,7 @@ type HeadInfo struct {
 	PodIP       string `json:"podIP,omitempty"`
 	ServiceIP   string `json:"serviceIP,omitempty"`
 	ServiceName string `json:"serviceName,omitempty"`
+	Ready       bool   `json:"ready"`
```

Comment:
I didn't add `omitempty` because `Ready: false` is also meaningful and should not be ignored.

Reply:
Should we consider a distinction between `Ready: nil` and "Ready: false`? `Ready: nil` is meaningful because it means "I don't know the status yet"

Reply:
Also consider that if we don't support `Ready: nil`, a future upgrade of kuberay will default `Ready` as `false` which can have unintended consequences

Reply:
I see, thank you! Also just noticed that https://pkg.go.dev/k8s.io/api/core/v1#ConditionStatus also has `unknown` status.